src/lib/vega-wallet/vega-wallet-service.ts

```diff
@@ -37,14 +37,22 @@ export interface DelegateSubmissionInput {
   };
 }
 
-export interface UndelegateSubmissionInput {
+export interface UndelegateEndOfEpochSubmissionInput {
```

Comment:
optional: can we make a enum for method and just have an undelgeate submission type rather than needing an interface per undelgation method

Reply:
Agree. We shouldn't be adding a whole new interface for every time a single field changes

Reply:
done